R/character.R

```diff
@@ -16,46 +16,72 @@
-#' @return Return a function. To make only one call you can perform a currying like in \code{lime(...)(...)}.
 #'
-#' TODO : add example
-#' TODO : for keep_word_position, make 2 versions of the text, one with _position and one without for the model to predict
+#' @examples
+#' \dontrun{
+#' # Explaining a model based on text data
+#'
+#' library(text2vec)
+#' library(lime)
+#' library(xgboost)
+#'
+#' data(train_sentences)
+#' data(test_sentences)
+#'
+#' get.matrix <- function(text) {
+#'   it <- itoken(text, progressbar = FALSE)
+#'   create_dtm(it, vectorizer = hash_vectorizer())
+#' }
+#'
+#' dtm_train = get.matrix(train_sentences$text)
 #'
-#' @importFrom purrr is_empty is_scalar_logical
-#' @importFrom stringdist seq_dist
+#' bst <- xgb.train(list(max_depth = 7, eta = 0.1, objective = "binary:logistic",
+#'                  eval_metric = "error", nthread = 1),
+#'                  xgb.DMatrix(dtm_train, label = train_sentences$class.text == "OWNX"),
+#'                  nrounds = 50)
+#'
+#' lime(test_sentences[5, text], bst, get.matrix, n_labels = 1,
+#'      number_features_explain = 2, keep_word_position = FALSE)()
+#' }
+#'
+#' @return Return a function. To make only one call you can perform a currying like in \code{lime(...)(...)}.
+#'
+#' @importFrom purrr is_empty is_scalar_logical is_null
 #' @importFrom magrittr %>%
 #' @importFrom assertthat validate_that not_empty is.scalar
 #' @export
-lime.character <- function(x, model, preprocess, tokenization = default_tokenize, keep_word_position = FALSE, kernel_width = 25,
+lime.character <- function(x, model, preprocess, tokenization = default_tokenize, keep_word_position = FALSE,
                            n_permutations = 5000, number_features_explain = 5, feature_selection_method = "auto",
                            labels = NULL, n_labels = NULL,  prediction = default_predict, ...) {
 
   validate_that("function" %in% class(preprocess))
   validate_that("function" %in% class(tokenization))
   validate_that("function" %in% class(prediction))
   validate_that(is_scalar_logical(keep_word_position))
-  validate_that(is.null(labels) + is.null(n_labels) == 1, msg = "You need to choose between labels and n_labels parameters.")
-  validate_that(!is.null(model))
+  validate_that(is_null(labels) + is.null(n_labels) == 1, msg = "You need to choose between labels and n_labels parameters.")
+  validate_that(!is_null(model))
   not_empty(x)
   validate_that(feature_selection_method %in% feature_selection_method())
-  validate_that(number_features_explain >= 1)
   is.scalar(number_features_explain)
-  validate_that(n_permutations >= 1)
+  validate_that(number_features_explain >= 1)
   is.scalar(n_permutations)
-  validate_that(kernel_width >= 1)
-  is.scalar(kernel_width)
+  validate_that(n_permutations >= 1)
```

Comment:
Didn't catch this the last time. `validate_that` should be substituted with `assert_that` and all assertions (e.g. `is.scalar`) should be wrapped in `assert_that` (they just return `TRUE`/`FALSE`)